### PR TITLE
refactor(obstacle_cruise_planner)!: add autoware_ prefix

### PR DESCRIPTION
**Pull request by yuki-takagi-66**
## Description
Add `autoware_` prefix to `obstacle_cruise_planner`.

Part of:
- https://github.com/autowarefoundation/autoware/issues/4569

## Related links
https://github.com/autowarefoundation/autoware/issues/4569

## Tests performed
works well in psim

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -51,7 +51,7 @@ control/autoware_external_cmd_selector/** fumiya.watanabe@tier4.jp satoshi.ota@t
 control/autoware_joy_controller/** fumiya.watanabe@tier4.jp satoshi.ota@tier4.jp shumpei.wakabayashi@tier4.jp taiki.tanaka@tier4.jp takamasa.horibe@tier4.jp takayuki.murooka@tier4.jp tomoya.kimura@tier4.jp
 control/autoware_lane_departure_checker/** kyoichi.sugahara@tier4.jp makoto.kurihara@tier4.jp
 control/mpc_lateral_controller/** takamasa.horibe@tier4.jp takayuki.murooka@tier4.jp
-control/obstacle_collision_checker/** fumiya.watanabe@tier4.jp satoshi.ota@tier4.jp shumpei.wakabayashi@tier4.jp taiki.tanaka@tier4.jp takamasa.horibe@tier4.jp takayuki.murooka@tier4.jp tomoya.kimura@tier4.jp
+control/autoware_obstacle_collision_checker/** fumiya.watanabe@tier4.jp satoshi.ota@tier4.jp shumpei.wakabayashi@tier4.jp taiki.tanaka@tier4.jp takamasa.horibe@tier4.jp takayuki.murooka@tier4.jp tomoya.kimura@tier4.jp
```
Comment (takayuki5168) [inline .github/CODEOWNERS:54]: different package

**Review by takayuki5168 — APPROVED** [review]
Other than a single comment, LGTM.

File: .github/CODEOWNERS (hunk 1 above)
Comment (yuki-takagi-66) [reply .github/CODEOWNERS]: Thank you

Comment (yuki-takagi-66) [reply .github/CODEOWNERS]: I've fixed

**Review by satoshi-ota — APPROVED** [review]

**Pull request merged**